superset-frontend/src/components/Menu/Menu.tsx

```diff
@@ -151,147 +258,116 @@ const StyledHeader = styled.header`
         ${({ theme }) => theme.gridUnit * 2}px;
     }
   }
+  .ant-menu-submenu-popup {
+    border-radius: 0px !important;
+    background-color: red !important;
+  }
 `;
 
+const { SubMenu } = DropdownMenu;
+
 export function Menu({
   data: { menu, brand, navbar_right: navbarRight, settings },
   isFrontendRoute = () => false,
 }: MenuProps) {
-  const [dropdownOpen, setDropdownOpen] = useState(false);
+  const [showMenu, setMenu] = useState<MenuMode>('horizontal');
 
-  return (
-    <StyledHeader className="top" id="main-menu">
-      <Navbar inverse fluid staticTop role="navigation">
-        <Navbar.Header>
-          <Navbar.Brand>
-            <a className="navbar-brand" href={brand.path}>
-              <img width={brand.width} src={brand.icon} alt={brand.alt} />
-            </a>
-          </Navbar.Brand>
-          <Navbar.Toggle />
-        </Navbar.Header>
-        <Nav data-test="navbar-top">
-          {menu.map((item, index) => {
-            const props = {
-              ...item,
-              isFrontendRoute: isFrontendRoute(item.url),
-              childs: item.childs?.map(c => {
-                if (typeof c === 'string') {
-                  return c;
-                }
+  useEffect(() => {
+    function handleResize() {
+      if (window.innerWidth <= 767) {
+        setMenu('inline');
+      } else setMenu('horizontal');
+    }
+    handleResize();
```

Comment:
debounce this... maybe with lodash, or @suddjian might have something to leverage...

Reply:
or, check out `useResizeDetector` from the Explore chart panel